ports/stm32/main.c

```diff
@@ -185,10 +185,14 @@ MP_NOINLINE static bool init_flash_fs(uint reset_mode) {
 
     if (len != -1) {
         // Detected a littlefs filesystem so create correct block device for it
-        mp_obj_t args[] = { MP_OBJ_NEW_QSTR(MP_QSTR_len), MP_OBJ_NEW_SMALL_INT(len) };
-        bdev = MP_OBJ_TYPE_GET_SLOT(&pyb_flash_type, make_new)(&pyb_flash_type, 0, 1, args);
+        mp_obj_t lfs_bdev = pyb_flash_new_obj(0, len);
+        if (lfs_bdev == mp_const_none) {
+            // Uncaught exception; len must be an invalid length.
+            mp_printf(&mp_plat_print, "MPY: corrupted filesystem\n");
+        } else {
+            bdev = lfs_bdev;
+        }
```

Comment:
I suggest to put this code in the above block loop, and then only bail out if the superblock can't be found anywhere.  Eg:
```c
for (size_t block_num = 0; block_num <= 1; ++block_num) {
    // look for lfs1 compatible superblock
    // look for lfs2 compatible superblock
    // if any superblock found, call pyb_flash_new_obj
    // if that succeeds, then break
}
// if no valid superblock was found, but littlefs string was found, emit the error message about corrupt filesystem
// otherwise fall through to attempt to mount the bdev
```